src/User/User.php

```diff
@@ -17,31 +17,12 @@ class User
     private const SESSION_AUTH_EXPIRE = '__auth_expire';
     private const SESSION_AUTH_ABSOLUTE_EXPIRE = '__auth_absolute_expire';
 
-    /**
-     * @var IdentityRepositoryInterface
-     */
-    private $identityRepository;
-
-    /**
-     * @var AccessCheckerInterface
-     */
-    private $accessChecker;
-
-    /**
-     * @var IdentityInterface
-     */
-    private $identity;
-
-    /**
-     * @var Session
-     */
-    private $session;
-
-    /**
-     * @var EventDispatcherInterface
-     */
-    private $eventDispatcher;
-
+    private IdentityRepositoryInterface $identityRepository;
+    private EventDispatcherInterface $eventDispatcher;
+    private ?AccessCheckerInterface $accessChecker = null;
+    private ?IdentityInterface $identity = null;
+    private ?Session $session = null;
```

Comment:
```suggestion
    private ?Session $session;
```
By default all properties have `null` value.

Reply:
no, in 7.4 if you use typing you need to initialize the property before it is called
if used like this it will throw an error when calling a property
`Fatal error: Uncaught Error: Typed property User::$session
must not be accessed before initialization`
now if the property can contain null you need to set it to null by default it is not set

Reply:
I didn't know that. Thank you